src/core/components/hv-route/index.tsx

```diff
@@ -577,12 +581,34 @@ export default function HvRoute(props: Types.Props) {
       // Use the beforeRemove event to remove the route from the stack
       const unsubscribeRemove: () => void = props.navigation.addListener(
         'beforeRemove',
-        () => {
-          NavigatorService.removeStackRoute(
-            docContext?.getDoc(),
-            props.route?.params?.url,
-            navigationContext.entrypointUrl,
-          );
+        (event: { preventDefault: () => void }) => {
+          // Use the current document state to access behaviors on the document
+          // Check for elements registered to interrupt back action via a trigger of BACK
+          const elements: Element[] | undefined = backContext?.get();
+          if (elements && elements.length > 0 && backContext?.onUpdate) {
+            // Process the elements
+            event.preventDefault();
+            elements.forEach(behaviorElement => {
+              const href = behaviorElement.getAttribute('href');
+              const action = behaviorElement.getAttribute('action');
+              if (backContext.onUpdate) {
+                backContext.onUpdate(href, action, behaviorElement, {
+                  behaviorElement,
+                  showIndicatorId: behaviorElement.getAttribute(
+                    'show-during-load',
+                  ),
+                  targetId: behaviorElement.getAttribute('target'),
+                });
+              }
```

Comment:
minor: to avoid having to test for `backContext?.onUpdate` twice, you can deconstruct / assign new constants which will satisfy TS:
```suggestion
          const { get, onUpdate } = backContext || {};
          const elements: Element[] | undefined = get && get();
          if (elements?.length > 0 && onUpdate) {
            // Process the elements
            event.preventDefault();
            elements.forEach(behaviorElement => {
              const href = behaviorElement.getAttribute('href');
              const action = behaviorElement.getAttribute('action');
              onUpdate(href, action, behaviorElement, {
                behaviorElement,
                showIndicatorId: behaviorElement.getAttribute(
                  'show-during-load',
                ),
                targetId: behaviorElement.getAttribute('target'),
              });
```